src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs

```diff
@@ -25,8 +25,20 @@ internal static class TcpClientExtensions
                 CancellationToken cancellationToken)
         {
             Exception error = null;
-            var remoteEndPoint = client.Client.RemoteEndPoint?.ToString();
-            var localEndPoint = client.Client.LocalEndPoint?.ToString();
+
+            var remoteEndPoint = string.Empty;
+            var localEndPoint = string.Empty;
+            try
+            {
+                remoteEndPoint = client.Client.RemoteEndPoint.ToString();
+                localEndPoint = client.Client.LocalEndPoint.ToString();
+            }
+            catch (SocketException socketException)
+            {
+                EqtTrace.Error(
+                        "TcpClientExtensions.MessageLoopAsync: failed to access endpoint of the socket {0}",
```

Comment:
"Failed to access the endpoint due to socket error : {0}"

Reply:
Why was the null check removed ?

Reply:
OK, fixed

Reply:
@foman Apologies for the misunderstanding.
We should keep the method signature. I expect the following.

TcpClientExtensions.MessageLoopAsync : Failed to access the endpoint due to socket error : {0}

Reply:
Ok